src/sentry/types/region.py

```diff
@@ -343,6 +346,21 @@ def find_regions_for_user(user_id: int) -> set[str]:
     return find_regions_for_orgs(org_ids)
 
 
+@control_silo_function
+def find_regions_for_sentry_app(sentry_app: SentryApp) -> set[str]:
+    from sentry.sentry_apps.models.sentry_app_installation import SentryAppInstallation
+
+    if SiloMode.get_current_mode() == SiloMode.MONOLITH:
+        return {settings.SENTRY_MONOLITH_REGION}
+
+    org_ids = set(
+        SentryAppInstallation.objects.filter(sentry_app=sentry_app).values_list(
```

Comment:
if this is a `set` and we're worried about pulling all of the org ids into Python, you can do this in one go with something like
`SentryAppInstallation.objects.filter(sentry_app=sentry_app).distinct('organization_id')` which will already be a set

Reply:
Dropping this here because I found this behavior really confusing and had to look this up, but `distinct(<fields>)` doesn't return a set, as it doesn't run a `SELECT DISTINCT(<columns>)` query. Instead, it runs a `SELECT DISTINCT ON(<fields>)` query specifically when using postgres: https://docs.djangoproject.com/en/5.1/ref/models/querysets/#django.db.models.query.QuerySet.distinct

So you still need a `values_list("region_name")` here 😅

Reply:
oh. i forgot this was about region name, i wrote this about the org_id part 😓